### PR TITLE
Update to Ruby 3.0

### PR DESCRIPTION
**Pull request by noelrappin**
The migration part of this was failing in Ruby 3.0 because of the keyword change in the enums.

Dropping support for Rails 5.1 and 5.2, which aren't compatible with Ruby 3.0.

Note that Rails 7 is also not supported because of a change in the Rails 7 active record internals, we'll catch that in another PR

### CODE REVIEW

File: sql_enum.gemspec [hunk 1]
```diff
@@ -25,8 +25,8 @@ Gem::Specification.new do |spec|
   spec.executables   = spec.files.grep(%r{^exe/}) { |f| File.basename(f) }
   spec.require_paths = ["lib"]
 
-  spec.add_dependency "activesupport", ">= 5.1"
-  spec.add_dependency "activerecord", ">= 5.1"
+  spec.add_dependency "activesupport", ">= 6.0", "< 7.0"
+  spec.add_dependency "activerecord", ">= 6.0", "< 7.0"
```
Comment (dpep) [inline sql_enum.gemspec:29]: could simplify to `~> 6`

File: spec/active_record/column_methods_spec.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    # no-op
+  end
+
+  include ActiveRecord::ConnectionAdapters::MySQL::ColumnMethods
+end
+
+module ActiveRecord
+  module ConnectionAdapters
+    module MySQL
+      RSpec.describe ColumnMethods do
```
Comment (dpep) [inline spec/active_record/column_methods_spec.rb:14]: why the weird nesting?

**Review by dpep — APPROVED** [review]
thanks for the upgrade!  since this is technically a breaking change, can you please bump the major version also?

File: spec/active_record/column_methods_spec.rb (hunk 2 above)
Comment (noelrappin) [reply spec/active_record/column_methods_spec.rb:14]: It's not weird. It's more robust than `RSpec.describe ActiveRecord::ConnectionAdapters::MySQL::ColumnMethods` against the possibility that the autoloader doesn't like something.

**Pull request merged**